tmff2: 0.82 -> 0.82-unstable-2025-02-03

RaySlash opened the pull request:
## Description

Linux kernel module for Thrustmaster T300RS, T248 and (experimental) TX, T128, T-GT II and TS-XW wheels 
> https://github.com/Kimplul/hid-tmff2/commit/343c01bcddf5368ef5465c9848c37bbe6ffc950d -> https://github.com/Kimplul/hid-tmff2/commit/a9312ead5720922e1c06b541ea48b6db3f289d36

- Fixes https://github.com/NixOS/nixpkgs/issues/379882

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

RaySlash committed:
tmff2: 0.82 -> 0.82-unstable-2025-02-03

343c01b → a9312ea

RaySlash commented:
Output of `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 380244"`
```console
these 2 paths will be fetched (0.13 MiB download, 0.67 MiB unpacked):
  /nix/store/17ph4i4lj423x62x49svxgz8174nlabl-nixpkgs-review-3.0.1
  /nix/store/8kj6zgs9za8dla8l9n9z3nvn3srhn5ss-python3.12-argcomplete-3.5.1
copying path '/nix/store/8kj6zgs9za8dla8l9n9z3nvn3srhn5ss-python3.12-argcomplete-3.5.1' from 'https://cache.nixos.org'...
copying path '/nix/store/17ph4i4lj423x62x49svxgz8174nlabl-nixpkgs-review-3.0.1' from 'https://cache.nixos.org'...
No GitHub token provided via GITHUB_TOKEN variable. Falling back to local evaluation.
Tip: Install the `gh` command line tool and run `gh auth login` to authenticate.
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/380244/merge:refs/nixpkgs-review/1
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (25/25), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 14 (delta 10), reused 3 (delta 0), pack-reused 0 (from 0)
Unpacking objects: 100% (14/14), 2.92 KiB | 747.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
   e23fa8bc2cea..2dcc7bb1cb40  master                 -> refs/nixpkgs-review/0
   3e7fb9550277..4b8fbea884cf  refs/pull/380244/merge -> refs/nixpkgs-review/1
$ git worktree add /home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs 2dcc7bb1cb408d160e6dd49abeef025a7c131a75
Preparing worktree (detached HEAD 2dcc7bb1cb40)
Updating files: 100% (46030/46030), done.
HEAD is now at 2dcc7bb1cb40 neovim-node-client: create binary in $out/bin (#380235)
Local evaluation for computing rebuilds
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs nixpkgs-overlays=/tmp/nix-shell-32733-0/tmp9cxjshk9 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git checkout 4b8fbea884cf6da9449d713860685442d78333b7
Previous HEAD position was 2dcc7bb1cb40 neovim-node-client: create binary in $out/bin (#380235)
HEAD is now at 4b8fbea884cf Merge 2b039b13727864e7cd34e76b2046d2f48ad74708 into 2dcc7bb1cb408d160e6dd49abeef025a7c131a75
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs nixpkgs-overlays=/tmp/nix-shell-32733-0/tmp9cxjshk9 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
--------- Impacted packages on 'x86_64-linux' ---------
23 packages updated:
hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2 hid-tmff2

warning: ignoring the client-specified setting 'system', because it is a restricted setting and you are not a trusted user

$ nix build --file /nix/store/17ph4i4lj423x62x49svxgz8174nlabl-nixpkgs-review-3.0.1/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs nixpkgs-overlays=/tmp/nix-shell-32733-0/tmp9cxjshk9' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr local-system x86_64-linux --argstr nixpkgs-path /home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs --argstr nixpkgs-config-path /tmp/nix-shell-32733-0/tmp5n92bypc.nix --argstr attrs-path /home/smj/.cache/nixpkgs-review/pr-380244/attrs.nix
warning: ignoring the client-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/380244

--------- Report for 'x86_64-linux' ---------
1 package marked as broken and skipped:
linuxKernel.packages.linux_5_4_hardened.hid-tmff2

20 packages built:
linuxKernel.packages.linux_5_10.hid-tmff2 linuxKernel.packages.linux_5_10_hardened.hid-tmff2 linuxKernel.packages.linux_5_15.hid-tmff2 linuxKernel.packages.linux_5_15_hardened.hid-tmff2 linuxKernel.packages.linux_5_4.hid-tmff2 linuxKernel.packages.linux_6_1.hid-tmff2 linuxKernel.packages.linux_6_11.hid-tmff2 linuxKernel.packages.linux_6_11_hardened.hid-tmff2 linuxKernel.packages.linux_6_12.hid-tmff2 linuxKernel.packages.linux_hardened.hid-tmff2 linuxKernel.packages.linux_6_13.hid-tmff2 linuxKernel.packages.linux_6_1_hardened.hid-tmff2 linuxKernel.packages.linux_6_6.hid-tmff2 linuxKernel.packages.linux_6_6_hardened.hid-tmff2 linuxKernel.packages.linux_latest_libre.hid-tmff2 linuxKernel.packages.linux_libre.hid-tmff2 linuxKernel.packages.linux_lqx.hid-tmff2 linuxKernel.packages.linux_xanmod.hid-tmff2 linuxKernel.packages.linux_xanmod_latest.hid-tmff2 linuxKernel.packages.linux_zen.hid-tmff2


$ /nix/store/cv7ah4ymfiplyvx64r9vd3wn054src41-nix-2.24.12/bin/nix-shell --argstr local-system x86_64-linux --argstr nixpkgs-path /home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs --argstr nixpkgs-config-path /tmp/nix-shell-32733-0/tmp5n92bypc.nix --argstr attrs-path /home/smj/.cache/nixpkgs-review/pr-380244/attrs.nix --nix-path 'nixpkgs=/home/smj/.cache/nixpkgs-review/pr-380244/nixpkgs nixpkgs-overlays=/tmp/nix-shell-32733-0/tmp9cxjshk9' /nix/store/17ph4i4lj423x62x49svxgz8174nlabl-nixpkgs-review-3.0.1/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix
these 3 paths will be fetched (1.70 MiB download, 10.93 MiB unpacked):
  /nix/store/6vpzxqaqvwiyqywssfavs1s51b4cgsin-bash-interactive-5.2p37
  /nix/store/x9lgx9pd242kw0sdvdmwvmgj6igw8h8k-ncurses-6.4.20221231
  /nix/store/hnnmxmq0dypban4ygm9xbd5d033nvgs1-readline-8.2p13
copying path '/nix/store/x9lgx9pd242kw0sdvdmwvmgj6igw8h8k-ncurses-6.4.20221231' from 'https://cache.nixos.org'...
copying path '/nix/store/hnnmxmq0dypban4ygm9xbd5d033nvgs1-readline-8.2p13' from 'https://cache.nixos.org'...
copying path '/nix/store/6vpzxqaqvwiyqywssfavs1s51b4cgsin-bash-interactive-5.2p37' from 'https://cache.nixos.org'...
```

keenanweaver reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 380244`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.hid-tmff2</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 19 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.hid-tmff2</li>
    <li>linuxKernel.packages.linux_5_10_hardened.hid-tmff2</li>
    <li>linuxKernel.packages.linux_5_15.hid-tmff2</li>
    <li>linuxKernel.packages.linux_5_15_hardened.hid-tmff2</li>
    <li>linuxKernel.packages.linux_5_4.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_1.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_11.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_11_hardened.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_12.hid-tmff2</li>
    <li>linuxKernel.packages.linux_hardened.hid-tmff2 (linuxKernel.packages.linux_6_12_hardened.hid-tmff2)</li>
    <li>linuxKernel.packages.linux_6_13.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_1_hardened.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_6.hid-tmff2</li>
    <li>linuxKernel.packages.linux_6_6_hardened.hid-tmff2</li>
    <li>linuxKernel.packages.linux_latest_libre.hid-tmff2</li>
    <li>linuxKernel.packages.linux_libre.hid-tmff2</li>
    <li>linuxKernel.packages.linux_lqx.hid-tmff2</li>
    <li>linuxKernel.packages.linux_xanmod.hid-tmff2 (linuxKernel.packages.linux_xanmod_latest.hid-tmff2, linuxKernel.packages.linux_xanmod_stable.hid-tmff2)</li>
    <li>linuxKernel.packages.linux_zen.hid-tmff2</li>
  </ul>
</details>

pbsds commented:
Is the version defined anywhere? Consider setting the version to `0.82-unstable-2025-02-04` and setting `unstableGitUpdater` as the `updateScript`

RaySlash commented:
> Is the version defined anywhere? Consider setting the version to `0.82-unstable-2025-02-04` and setting `unstableGitUpdater` as the `updateScript`

This is not a version bump from the driver maintainer side, rather a patch to address a change in `6.12` kernel version. Also, I did not get the `unstableGitUpdater` part. Could you suggest changes inline please?

pbsds commented:
Regarding the version schema, please view https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning

For the updater, I assumed you wanted automatic updates from the source repo by r-ryantm. For this to happen, setting `passthru.updateScript = unstableGitUpdater { };` is a good start, perhaps some more flags are needed

RaySlash commented:
> Regarding the version schema, please view https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning

Thank you. Just updated the version scheme according to the guidelines.

> For the updater, I assumed you wanted automatic updates from the source repo by r-ryantm. For this to happen, setting `passthru.updateScript = unstableGitUpdater { };` is a good start, perhaps some more flags are needed

Interesting, I added the line without any flags. I was not able to get much info regarding `unstableGitUpdater` as the docs for `nixpkgs-update` do not really mention it. I believe it is good as it is for now and can see if the update fails from logs for r-ryantm.
